src/main/java/com/google/sps/servlets/AvailabilityServlet.java

```diff
@@ -28,10 +28,32 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.sps.utilities.RealAvailabilityDatastore;
+import com.google.sps.utilities.MockAvailabilityDatastore;
+import com.google.sps.utilities.AvailabilityDatastoreService;
 
+/** Servlet that getting and adding availability for a tutor. */
 @WebServlet("/availability")
 public class AvailabilityServlet extends HttpServlet {
 
+    private AvailabilityDatastoreService datastore;
+
+    /**
+    * Because we created a constructor with a parameter (the testing one), the default empty constructor does not work anymore so we have to explicitly create it. 
+    * We need the default one for deployment because the servlet is created without parameters.
+    */
+    public AvailabilityServlet(){}
+
+    public AvailabilityServlet(boolean test) {
+        if(test) {
+            datastore = new MockAvailabilityDatastore();
+        }
+    }
+
+    public void init() {
+        datastore = new RealAvailabilityDatastore();
+    }
+
```

Comment:
Why do we need to initialize a new one here?

Reply:
The variable `datastore` is not initialized before this so it gets initialized in the `init` method with the real datastore when the servlet is created. The tests don't call `init`, so `datastore` gets initialized in the test constructor with the mock implementation.